rsconnect/api.py

```diff
@@ -981,6 +991,21 @@ def make_deployment_name(self, title, force_unique):
 
         return name
 
+    def list_runtime_caches(self):
+        return self.client.system_caches_runtime_list()
+
+    def delete_runtime_cache(self, language, version, image_name, dry_run):
+        target = {"language": language, "version": version, "image_name": image_name, "dry_run": dry_run}
+        result = self.client.system_caches_runtime_delete(target)
+        if result["task_id"] is None:
+            print("Dry run finished")
+            return result
+        else:
+            (log_lines, task_status) = self.client.wait_for_task(
+                result["task_id"], connect_logger.info, raise_on_error=False
+            )
+            return task_status
+
```

Comment:
Executor functions should be chainables that ends with `return self`.

Reply:
Gotcha. I understand that information like the list function above should be a property. With this, I should just return `self` at the end of the function and no other changes? Or does it make sense to surface the final task result somehow? I guess I could pass verbosity down into this function.

Reply:
Yeah, a `return self` here seems right. `wait_for_task` is printing out the logs so the thing left to do is decide whether you want to add any custom exceptions based on the task_status in the `else` block. Or just pass `raise_on_error=True`

```suggestion
    def delete_runtime_cache(self, language, version, image_name, dry_run):
        target = {"language": language, "version": version, "image_name": image_name, "dry_run": dry_run}
        result = self.client.system_caches_runtime_delete(target)
        if result["task_id"] is None:
            print("Dry run finished")
        else:
            (log_lines, task_status) = self.client.wait_for_task(
                result["task_id"], connect_logger.info, raise_on_error=False
            )
        return self
```